lib/FrontendTool/FrontendTool.cpp

```diff
@@ -791,71 +909,19 @@ static bool performCompile(CompilerInstance &Instance,
     if (!opts.ObjCHeaderOutputPath.empty())
       return printAsObjC(opts.ObjCHeaderOutputPath, Instance.getMainModule(),
                          opts.ImplicitObjCHeaderPath, moduleIsPublic);
-    if (shouldIndex) {
-      if (emitIndexData(Instance.getPrimarySourceFile(),
-                        Invocation, Instance))
-        return true;
-    }
-    return Context.hadError();
-  }
 
-  auto &SILOpts = Invocation.getSILOptions();
-  if (!opts.TBDPath.empty()) {
-    auto installName = opts.TBDInstallName.empty()
-                           ? "lib" + Invocation.getModuleName().str() + ".dylib"
-                           : opts.TBDInstallName;
-
-    if (writeTBD(Instance.getMainModule(), SILOpts.hasMultipleIGMs(),
-                 opts.TBDPath, installName))
-      return true;
+    return emitIndexDataIfNeeded(Instance.getPrimarySourceFile(), Invocation,
```

Comment:
The old code here is kind of weird—it assumes it's *either* indexing *or* printing an ObjC header. That's not really a good thing; you should be able to do both.

Reply:
I was wondering about that:) Fixed to try to do both, as needed, and report the disjunction of the hadError's.